modules/postgres/postgres.go

```diff
@@ -170,6 +170,98 @@ func WithSnapshotName(name string) SnapshotOption {
 	}
 }
 
+// WithSSLSettings configures the Postgres server to run with the provided CA Chain
+// This will not function if the corresponding postgres conf is not correctly configured.
+// Namely the paths below must match what is set in the conf file
+func WithSSLSettings(sslSettings SSLSettings) testcontainers.CustomizeRequestOption {
+	const postgresCaCertPath = "/tmp/data/ca_cert.pem"
+	const postgresCertPath = "/tmp/data/server.cert"
+	const postgresKeyPath = "/tmp/data/server.key"
+
+	const defaultPermission = 0o600
+
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Files = append(req.Files, testcontainers.ContainerFile{
+			HostFilePath:      sslSettings.CACertFile,
+			ContainerFilePath: postgresCaCertPath,
+			FileMode:          defaultPermission,
+		})
+		req.Files = append(req.Files, testcontainers.ContainerFile{
+			HostFilePath:      sslSettings.CertFile,
+			ContainerFilePath: postgresCertPath,
+			FileMode:          defaultPermission,
+		})
+		req.Files = append(req.Files, testcontainers.ContainerFile{
+			HostFilePath:      sslSettings.KeyFile,
+			ContainerFilePath: postgresKeyPath,
+			FileMode:          defaultPermission,
+		})
+
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForLog("database system is ready to accept connections"))
+
+		internalEntrypoint(req)
+	}
+}
+
+func internalEntrypoint(req *testcontainers.GenericContainerRequest) {
+
+	const entrypointPath = "/usr/local/bin/docker-entrypoint-ssl.bash"
+
+	entrypoint := `
+	#!/usr/bin/env bash
+	set -Eeo pipefail
+
+
+	pUID=$(id -u postgres)
+	pGID=$(id -g postgres)
+
+	if [ -z "$pUID" ]
+	then
+      exit 1
```

Comment:
Not sure why github is rendering it like this. In code the `exit 1` is correctly indented.

Reply:
Have you checked spaces Vs tabs?? 🤔